store/postgres/src/subgraph_store.rs

```diff
@@ -499,6 +513,8 @@ impl SubgraphStoreInner {
         #[cfg(not(debug_assertions))]
         assert!(!replace);
 
+        self.evict(&schema.id)?;
```

Comment:
Is the reason we call `evict` on deployment creation so we have more confidence that the cache will be cleared?

Reply:
Yes, to make sure that we don't have a stale entry for the same hash in the cache (which could happen if a previous deployment of the same hash was stopped and then redeployed) This is really redundant with the eviction in `stop_subgraph` but since it's a cheap operation seemed safer to do it in both places.